src/graphql/apollo.js

```diff
@@ -1,5 +1,68 @@
 import { ApolloClient } from 'apollo-boost'
+import { InMemoryCache } from 'apollo-cache-inmemory'
 
-const client = new ApolloClient({ uri: 'http://localhost:4000/graphql' })
+import { HttpLink } from 'apollo-link-http'
+import { onError } from 'apollo-link-error'
+import { ApolloLink, from, Observable } from 'apollo-link'
+
+class AuthLink extends ApolloLink {
+  /* eslint-disable-next-line */
+  handleLogout(operation) {
+    const { authExpired, response } = operation.getContext()
+
+    if (response && response.redirected) {
+      const { url } = response
+      /* eslint-disable-next-line */
+      console.log(`The redirected url is ${url}`)
+    }
+
+    if (authExpired) {
+      /* eslint-disable-next-line */
+      console.log('logged out')
+    }
+  }
+
+  request(operation, forward) {
+    return new Observable((obs) => {
+      forward(operation).subscribe({
+        next: (data) => {
+          this.handleLogout(operation)
+          obs.next(data)
+        },
+        error: (err) => {
+          this.handleLogout(operation)
+          obs.next(err)
+        },
+        complete: obs.complete.bind(obs),
+      })
+    })
+  }
+}
+
+const auth = new AuthLink()
+const error = onError(({ operation, networkError, graphQLErrors }) => {
+  const { response } = operation.getContext()
+  // this is when you have mixed data + errors
+  if (response && response.status === 401) {
+    operation.setContext({ authExpired: true })
+  }
+
+  // this is from non mixed data or execution failure
+  // i.e. { errors } where no { data } field is present
+  if (networkError && networkError.statusCode === 401) {
+    operation.setContext({ authExpired: true })
+  }
+})
+
+const http = new HttpLink({
+  uri: 'https://gql.now.sh/',
+})
+
+const link = from([auth, error, http])
+
+const client = new ApolloClient({
+  link,
+  cache: new InMemoryCache(),
```

Comment:
Do we really need explicitly define this? apollo-boost readme says it does these things by default. https://github.com/apollographql/apollo-client/blob/master/packages/apollo-boost/README.md
`link` can be set in `uri` option and `cache` has default value `InMemoryCache` from `apollo-cache-inmemory` [Link](https://www.apollographql.com/docs/react/essentials/get-started.html#configuration)

Reply:
Let me check it later